lib/portage/data.py

```diff
@@ -178,6 +178,14 @@ def _get_global(k):
 				noiselevel=-1)
 			portage_group_warning()
 
+		portage_homedir_exists = os.access(
+				first_existing(os.path.expanduser("~portage")), portage.os.W_OK)
```

Comment:
I don't understand the purpose of the `first_existing()` call here.

Reply:
I was copying a pattern I'd seen elsewhere to check for existing directories and write access. 
Looking at it again, I suppose in this case `first_existing()` might lead to a false positive if portage was being run as root. I'll change it now.